src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1121,3 +1121,25 @@ def as_user_validator(namespace):
             import argparse
             raise argparse.ArgumentError(
                 None, "incorrect usage: specify '--auth-mode login' when as-user is enabled")
+
+
+def validator_delete_retention_days(namespace):
+    if namespace.enable_delete_retention is True and namespace.delete_retention_days is None:
+        raise ValueError(
+            "incorrect usage: specify '--delete-retention-days' when '--enable-delete-retention' is enabled")
+
+    if namespace.enable_delete_retention is False and namespace.delete_retention_days is not None:
+        raise ValueError(
+            "incorrect usage: can't set '--delete-retention-days' when '--enable-delete-retention' is disabled")
```

Comment:
the value for --delete-retention-days is invalid when --enable-delete-retention is set to false